evalml/data_checks/uniqueness_data_check.py

```diff
@@ -0,0 +1,105 @@
+from evalml.data_checks import (
+    DataCheck,
+    DataCheckMessageCode,
+    DataCheckWarning
+)
+from evalml.problem_types import (
+    handle_problem_types,
+    is_multiclass,
+    is_regression
+)
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper
+)
+
+warning_not_unique_enough = "Input columns ({}) for {} problem type are not unique enough."
+warning_too_unique = "Input columns ({}) for {} problem type are too unique."
+
+
+class UniquenessDataCheck(DataCheck):
+    """Checks if there are any columns in the input that are either too unique for categorical problems
+    or not unique enough for regression problems."""
+
+    def __init__(self, problem_type, threshold=0.50):
+        """Checks each column in the input to determine the uniqueness of the values in those columns.
+
+        Arguments:
+            problem_type (str or ProblemTypes): The specific problem type to data check for.
+                e.g. 'binary', 'multiclass', 'regression, 'time series regression'
+            threshold(float): The threshold to set as either an upper or lower bound on uniqueness.
```

Comment:
Can we explain what this threshold means for different cases here? ie, If I was looking at a regression problem, would I want to set the threshold higher or lower for checking uniqueness?